### DISCUSSION TITLE
Derive route from interface name and/or operation names.

### DISCUSSION THREAD

**Discussion by boan-anbo** [microsoft/typespec#4675] [Q&A] [answered]
Coming from GRPC, I wonder if there is any decorators that one can use to automatically derive route path by the name of the operation?

```typescript

@route("/someservice") // regular route 
interface SomeService {

    @autoRoute // mapped to '/someservice/create'
    op create(): Response;  
    
}

// Or when applied to interface/namespace

@autoRoute //  mapped to '/someservice'
interface SomeService {

   op create(): Response; // mapped to  '/someservice/create'
}

```

Is there any ways to achieve something similar to this? Thank you!

**Comment by timotheeguerin** [comment, marked as answer]
You can use `@action` for this from the rest library

[Playground](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdCI7CgpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKfSkKbmFtZXNwYWNlIERlbW%2FHGzsKCnVzaW5nIFR5cGVTcGVjLkh0dHA70BVSZXN0xGxhdXRvUm91dGUKQHLEBygicGV0cyIpCmludGVyZsRdxnvHXyDkAJVAYWN0aW9uCiAgZG8oKTogdm9pZDsKfQo%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D)

**Reply by boan-anbo** [reply]
Awesome! Thank you
